qutebrowser/mainwindow/prompt.py

```diff
@@ -467,6 +467,27 @@ def prompt_yank(self, sel=False):
         utils.set_clipboard(question.url, sel)
         message.info("Yanked to {}: {}".format(target, question.url))
 
+    @cmdutils.register(
+        instance='prompt-container', scope='window',
+        modes=[usertypes.KeyMode.prompt])
+    def prompt_external_picker(self):
+        """Choose a location using a configured external picker."""
+        picker = config.val.fileselect.folder.command
+        if not picker:
+            log.webview.warning(
+                "No external location picker (fileselect.folder.command) configured."
+            )
+            return
+        folders = shared.choose_file(shared.FileSelectionMode.folder)
+        if not folders:
+            message.info("No folder chosen.")
+            return
+        assert self._prompt is not None
+        question = self._prompt.question
+        # choose_file already checks that this is max one folder
+        question.default = folders[0]
+        self._prompt._lineedit.setText(question.default)
```

Comment:
This should instead be a method in `_BasePrompt` raising `UnsupportedOperationError`, which is then overridden in `LineEditPrompt`. Right now I suspect that this would crash with an `AttributeError` if this command is run with e.g. an authentication or yes/no prompt (JS confirm, TLS error, etc.).

I think the method can be kept general (i.e. `set_text` rather than `set_path` or somesuch), which would also allow implementing an equivalent to `:set-cmd-text` for prompt mode later.

Alternatively, we could just call `accept` here as you mention, I'll answer to that in my main comment.

Reply:
Changed to directly accept in 65c3007fc07f708e716c9cb14fefd10bf5f240d7, is that as simple as calling `self._prompt.accept` with the path to the folder?